### PR TITLE
appveyor build with preparation for automatic github deployment

### PR DESCRIPTION
**Pull request by chcg**
- added appveyor.yml with preparation for automatic github deployment on tagging (needs secure token at the TODO, see also https://www.appveyor.com/docs/deployment/github/#provider-settings how to generate Personal API access token at https://github.com/settings/tokens and encrypt it)
- avoid vs2015 compiler warnings about hidden global vars
- enabled multithreaded builds in vcxproj
- enabled sdl checks in vcxproj
- corrected platform toolset for x64 debug build
- corrected copy action of readme.txt

@donho Test build see https://ci.appveyor.com/project/chcg/pork2sausage/builds/35802809

### CODE REVIEW

**Comment by chcg** [comment]
@donho Update for ARM64, see https://ci.appveyor.com/project/chcg/pork2sausage/builds/38926060. Didn't test it further apart that it could be compiled. Probably some other parts might need to be modernized also, like the java jre path.

**Comment by donho** [comment]
@chcg 
> corredted copy action of readme.txt

I didn't see this part. Is it missing or the above log is not accurate?

Also, is it necessary for the change of variable names in both `Pork2Sausage.h` & `Pork2Sausage.cpp`?

File: vs.proj/pork2sausage.vcxproj [hunk 1]
```diff
@@ -151,22 +207,22 @@
     </Link>
     <PostBuildEvent>
       <Command>copy ..\license.txt ..\bin\license.txt
-copy ..\readme.txt ..\bin\readmetxt
```
Comment (chcg) [inline vs.proj/pork2sausage.vcxproj:154]: @donho readmetxt is corrected to readme.txt here and below

**Comment by chcg** [comment]
> Also, is it necessary for the change of variable names in both Pork2Sausage.h & Pork2Sausage.cpp?

As TreatWarningAsError is true the compiler warnings about hidden global var either need to be fixed by renaming the local vars or by switching of this specific warning I'm afraid.

File: src/Pork2Sausage.h [hunk 2]
```diff
@@ -96,9 +96,9 @@ struct CmdParam {
 
 
 
-void launchProgram(const CmdParam & cmdParam);
+void launchProgram(const CmdParam & cmdParamValue);
 
-int getCmdsFromConf(const TCHAR *confPath, CmdParam *cmdParam, int maxNbCmd);
+int getCmdsFromConf(const TCHAR *confPathValue, CmdParam *cmdParamValue, int maxNbCmd);
```
Comment (donho) [inline src/Pork2Sausage.h:101]: @chcg 
I was talking about changing the argument variable name - it's not necessary at all.

Comment (chcg) [reply src/Pork2Sausage.h:101]: Just reverted this changes locally and still get:

1>------ Build started: Project: pork2sausage, Configuration: Release ARM64 ------
1>Pork2Sausage.cpp
1>Process.cpp
1>c:\v141\pork2sausage\src\pork2sausage.cpp(195): error C2220: warning treated as error - no 'object' file generated
1>c:\v141\pork2sausage\src\pork2sausage.cpp(195): warning C4459: declaration of 'confPath' hides global declaration
1>c:\v141\pork2sausage\src\pork2sausage.cpp(39): note: see declaration of 'confPath'
1>c:\v141\pork2sausage\src\pork2sausage.cpp(195): warning C4459: declaration of 'cmdParam' hides global declaration
1>c:\v141\pork2sausage\src\pork2sausage.cpp(41): note: see declaration of 'cmdParam'
1>c:\v141\pork2sausage\src\pork2sausage.cpp(254): warning C4459: declaration of 'cmdParam' hides global declaration
1>c:\v141\pork2sausage\src\pork2sausage.cpp(41): note: see declaration of 'cmdParam'
1>Done building project "pork2sausage.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========

Comment (donho) [reply src/Pork2Sausage.h:101]: Thank you for the explain.

**Comment by donho** [comment]
@chcg 
FYI:
I just got the crash of Notepad++ v8 & v7.9.5 with pork2sausage (not due to your PR).
I'll remove it from the Plugin List temporarily before fixing this crash.

**Pull request closed without merging**